nixos/modules/services/web-apps/miniflux.nix

```diff
@@ -0,0 +1,123 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.miniflux;
+
+  dbUser = "miniflux";
+  dbPassword = "miniflux";
+  dbHost = "localhost";
+  dbName = "miniflux";
+in
+
+{
+  options = {
+    services.miniflux = {
+      enable = mkEnableOption "miniflux";
+
+      user = mkOption {
+        type = types.str;
+        default = "miniflux";
+        description = ''
+          User account under which miniflux runs.
+        '';
+      };
+
+      group = mkOption {
+        type = types.str;
+        default = "miniflux";
+        description = ''
+          User group under which miniflux runs.
+        '';
+      };
+
+      listenAddress = mkOption {
+        type = types.str;
+        default = "localhost";
+        description = ''
+          The host the daemon will listen on.
+        '';
+      };
+
+      listenPort = mkOption  {
+        type = types.int;
+        default = 8080;
+        description = ''
+          The port the daemon will listen on.
+        '';
+      };
+
+      adminUser = mkOption  {
+        type = types.str;
+        default = "admin";
+        description = ''
+          Miniflux admin username.
+        '';
+      };
+
+      adminPassword = mkOption {
+        type = types.str;
+        default = "password";
+        description = ''
+          Miniflux admin password.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    services.postgresql.enable = true;
+
+    users = {
+      users.miniflux = {
+        name = cfg.user;
+        group = cfg.group;
+        description = "miniflux service user";
+        isSystemUser = true;
+        createHome = false;
+      };
+      groups.miniflux = {
+        name = cfg.group;
+      };
+    };
+
+    systemd.services.miniflux = {
+      description = "Miniflux service";
+      wantedBy = [ "multi-user.target" ];
+      requires = [ "postgresql.service" ];
+      after = [ "network.target" "postgresql.service" ];
+
+      serviceConfig = {
+        PermissionsStartOnly = true;
+        PrivateTmp = false;
+        ExecStart = "${pkgs.miniflux}/bin/miniflux";
+        User = cfg.user;
+        Group = cfg.group;
+      };
+
+      environment = {
+        LISTEN_ADDR = "${cfg.listenAddress}:${toString cfg.listenPort}";
+        DATABASE_URL = "postgresql://${dbUser}:${dbPassword}@${dbHost}/${dbName}?sslmode=disable";
+        RUN_MIGRATIONS = "1";
+        CREATE_ADMIN = "1";
+        ADMIN_USERNAME = "${cfg.adminUser}";
+        ADMIN_PASSWORD = "${cfg.adminPassword}";
```

Comment:
Would you be open to modifying the code to not store plaintext passwords in the Nix store? Instead of this approach you could generate an [EnvironmentFile](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#EnvironmentFile=) in your preStart script which is only readable by your miniflux user and tell the service to use this generated file instead of storing your passwords in the environment variables defined in the world readable service file.

Reply:
It has be done as you asked.

Reply:
Awesome!